router_client.go

```diff
@@ -883,6 +886,56 @@ type MissionControlEntry struct {
 	SuccessAmt lnwire.MilliSatoshi
 }
 
+// QueryMissionControl will query Mission Control state from lnd.
+func (r *routerClient) QueryMissionControl(ctx context.Context) (
+	[]MissionControlEntry, error) {
+
+	rpcCtx, cancel := context.WithTimeout(ctx, r.timeout)
+	defer cancel()
+
+	req := &routerrpc.QueryMissionControlRequest{}
+	res, err := r.client.QueryMissionControl(
+		r.routerKitMac.WithMacaroonAuth(rpcCtx), req,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]MissionControlEntry, 0, len(res.Pairs))
+	for _, pair := range res.Pairs {
+		if pair.History == nil {
+			continue
+		}
+
+		nodeFrom, err := route.NewVertexFromBytes(pair.NodeFrom)
+		if err != nil {
+			return nil, err
+		}
+
+		nodeTo, err := route.NewVertexFromBytes(pair.NodeTo)
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, MissionControlEntry{
+			NodeFrom: nodeFrom,
+			NodeTo:   nodeTo,
+			FailTime: time.Unix(pair.History.FailTime, 0),
```

Comment:
Do we need a `!=0` check here? I think we can have entries which only have one timestamp, so we'll send up with results in 1970?

Reply:
Yes indeed, very good catch! Fixed.